run_helper_scripts/extrapolate_ddgs.py

```diff
@@ -0,0 +1,187 @@
+            year_columns.append(int(column))
+        else:
+            index_columns.append(column)
+
+    year_columns = sorted(year_columns)
+    LOG.debug("Found following years in DDG: %s", ", ".join(str(i) for i in year_columns))
+    ddg = ddg.set_index(index_columns)
+    ddg.columns = pd.to_numeric(ddg.columns, downcast="unsigned")
+
+    # Rename any unnamed index columns
+    ddg.index.names = ["" if n.lower().startswith("unnamed") else n for n in ddg.index.names]
+
+    return ddg, year_columns
+
+
+def _check_calculation_years(ddg_years: list[int], calculation_years: int) -> tuple[int, int]:
+    if calculation_years > len(ddg_years):
+        raise ValueError(
+            f"{len(ddg_years)} years found in DDG so cannot use "
+            f"{calculation_years} years for extrapolation calculation"
+        )
+    if calculation_years < -1 or calculation_years == 0:
+        raise ValueError(
+            "number of calculation years should be a positive integer "
+            f"(or -1 to select all years) not {calculation_years}"
+        )
+
+    if calculation_years == -1:
+        calc_range = (ddg_years[0], ddg_years[-1])
+    else:
+        calc_range = (ddg_years[-1 * calculation_years], ddg_years[-1])
+    LOG.info("Using %s - %s for calculating extrapolation factor", *calc_range)
+    return calc_range
+
+
+def extrapolate_ddg(
+    input_file: pathlib.Path,
+    output_file: pathlib.Path,
+    extrapolation_year: int,
+    calculation_years: int,
+) -> None:
+    """Extrapolate give `input_file` to `extrapolation_year`.
+
+    Won't perform extrapolation if the `input_file` already contains
+    a year column >= `extrapolation_year`. Adds all years between
+    the maximum current year in the file and `extrapolation_year`.
+
+    Parameters
+    ----------
+    input_file : pathlib.Path
+        Path to the DDG file.
+    output_file : pathlib.Path
+        Path to save extrapolated file to.
+    extrapolation_year : int
+        Year to extrapolate to, inclusive.
+    calculation_years : int
+        Number of years to use when calculating yearly average increase.
+    """
+    LOG.info("Reading %s", input_file.name)
+    ddg = pd.read_csv(input_file)
+    ddg, year_columns = _check_columns(ddg)
+
+    if extrapolation_year <= year_columns[-1]:
+        LOG.info(
+            "Extrapolation aborted because DDG %s max year "
+            "is %s which is larger than the extrapolation year %s",
+            input_file.stem,
+            year_columns[-1],
+            extrapolation_year,
+        )
+        return
+
+    calc_range = _check_calculation_years(year_columns, calculation_years)
+
+    yearly_increase = (ddg[calc_range[1]] - ddg[calc_range[0]]) / (
+        calc_range[1] - calc_range[0]
+    )
+    yearly_increase.name = "Average Yearly Increase"
+    increase_path = output_file.with_name(output_file.stem + "-average_yearly_increase.csv")
+    yearly_increase.to_csv(increase_path)
```

Comment:
Is this the yearly increase "report" looking thing being output? Could all of these be placed into a subfolder to make it easier to select all the DDGs I want and copy / paste them where I need them?

I've manually done this in the outputs already if you want a point of reference